### ISSUE TITLE
Worms:Battle Islands graphic problem

### ISSUE THREAD

**Issue by kdhslss** [hrydgard/ppsspp#18713]
### Game or games this happens in

NPUH-10045-Worms:Battle Islands

### What area of the game

The graphic in the square of the War Room screen isn't shown.(Main menu→Missions→Campaign→War Room)
![Screenshot_2024-01-17-12-48-03-1413888490_EDIT_1](https://github.com/hrydgard/ppsspp/assets/87033053/30b0f836-a3dc-4be0-9f6b-ec439fb58049)

### What happens

I've tried the latest version of the emulator and changed the graphic backend but the problem still can't be fixed.

### What should happen

....

### GE frame capture

...

### Platform

Android

### Mobile device model or graphics card (GPU)

mali 

### PPSSPP version affected

v1.16.6

### Last working version

None

### Graphics backend (3D API)

Vulkan

### Checklist

- [X] Test in the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already fixed.
- [X] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [X] Try resetting settings or older versions and include if the issue is related.
- [x] Try changing graphics settings to determine if one causes the glitch (especially speed hacks or enhancements/replacements.)
- [X] Include logs or screenshots of issue.

**Comment by sum2012** [comment]
@kdhslss do you enable graphic of any "Speed hack" option ?

**Comment by kdhslss** [comment]
@sum2012 Well, if you don't enable "Speed hack" option it still looks the same. Besides, if you enable "Software rending" mode, the graphic in the frame still looks abnormal.
![Screenshot_2024-01-23-12-48-57-1966847027_EDIT_1](https://github.com/hrydgard/ppsspp/assets/87033053/717b2b58-7301-4d93-ab12-5f2f97c5cd82)
![Screenshot_2024-01-23-12-48-44-1602798418_EDIT_1](https://github.com/hrydgard/ppsspp/assets/87033053/d9b60ff3-1ab5-4fbb-b2ca-2be01f92247b)

**Comment by unknownbrackets** [comment]
Is it possible to see this problem in a demo for one of the Worms games?

If it's wrong even in software rendering, there's probably something deeper going awry...

-[Unknown]

**Comment by sum2012** [comment]
@kdhslss I cannot see "War Room" in the screen
![3](https://github.com/hrydgard/ppsspp/assets/2177532/60e985b0-5683-47bd-a1fb-934033f40a94)

**Comment by hrydgard** [comment]
> @sum2012 Well, if you don't enable "Speed hack" option, it still looks the same. Besides, if you enable "Software rendering" mode, the graphic in the frame still looks abnormal.

Hm, are you sure that's abnormal? Seems like a sensible thing to show the shape of the map before you play without giving up too much other information. Better check on hardware. Though, it's not exactly pretty, I give you that.

**Comment by kdhslss** [comment]
@sum2012 When you are in "Campaign" screen, press "X" botton to see "Mission Briefing" screen, in that screen select "Start" option and "War Room" screen will appear.
![Screenshot_2024-01-30-11-41-19-1727875718_EDIT_1](https://github.com/hrydgard/ppsspp/assets/87033053/2ec29ec5-43ee-4830-8aaf-156ecd3f1696)

**Comment by kdhslss** [comment]
> > @sum2012 Well, if you don't enable "Speed hack" option, it still looks the same. Besides, if you enable "Software rendering" mode, the graphic in the frame still looks abnormal.
> 
> Hm, are you sure that's abnormal? Seems like a sensible thing to show the shape of the map before you play without giving up too much other information. Better check on hardware. Though, it's not exactly pretty, I give you that.

Yes, you're right. The graphic in "Software Rendering" mode is correct.

**Comment by kdhslss** [comment]
@hrydgard Yes, you're right. The graphic in "Software Rendering" mode is correct.

**Comment by hrydgard** [comment]
Well it's still broken in hw so I'll reopen.

**Comment by sum2012** [comment]
I am using  "Software rendering" get bad graphic
Attach gamefaq save
[worms-battle-islands.22355.zip](https://github.com/hrydgard/ppsspp/files/14097887/worms-battle-islands.22355.zip)

![3](https://github.com/hrydgard/ppsspp/assets/2177532/dfd3219a-eb67-42ec-8bf7-9c0cbf8eca0a)

**Comment by hrydgard** [comment]
Great to confirm that. However, also very strange that the software renderer failed for you but worked for kdhslss.

**Comment by hrydgard** [comment]
Oh, good find!

**Comment by sum2012** [comment]
In the newest windows build fixed software render
![3](https://github.com/hrydgard/ppsspp/assets/2177532/3e41346f-ef6c-40bc-a798-b1f5b6c36418)

software frame dump:
[NPUH10045-softgpu.zip](https://github.com/hrydgard/ppsspp/files/14125133/NPUH10045-softgpu.zip)

Vulkan frame dump:
[NPUH10045-vulkan.zip](https://github.com/hrydgard/ppsspp/files/14125138/NPUH10045-vulkan.zip)

**Comment by hrydgard** [comment]
That's strange, no clue what change might have fixed it...
